source/source.go

```diff
@@ -0,0 +1,227 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/conduitio-labs/conduit-connector-redshift/config"
+	"github.com/conduitio-labs/conduit-connector-redshift/source/iterator"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/multierr"
+
+	_ "github.com/jackc/pgx/v5/stdlib" // sql driver
+)
+
+const (
+	// driverName is a database driver name.
+	driverName = "pgx"
+	// keySearchPath is a key of get parameter of a datatable's schema name.
+	keySearchPath = "search_path"
+
+	// querySelectLastProcessedValFmt is a query pattern to select the last value of the orderingColumn column.
+	querySelectLastProcessedValFmt = "SELECT %s FROM %s ORDER BY %s DESC LIMIT 1;"
+)
+
+// Iterator interface.
+type Iterator interface {
+	HasNext(context.Context) (bool, error)
+	Next(context.Context) (sdk.Record, error)
+	Stop() error
+}
+
+// Source is an Amazon Redshift source plugin.
+type Source struct {
+	sdk.UnimplementedSource
+
+	config           config.Source
+	db               *sqlx.DB
```

Comment:
Could we remove it and use `db` from `iterator`?

Reply:
Done.